src/hb-ot-post-table-v2subset.hh

```diff
@@ -79,6 +112,7 @@ HB_INTERNAL bool postV2Tail::subset (hb_subset_context_t *c) const
   post::accelerator_t _post;
   _post.init (c->plan->source);
 
+  hb_hashmap_t<unsigned, hb_set_t *, (unsigned)-1, nullptr> len_to_gids;
```

Comment:
Make this a map from hb_bytes_t* to the index of the string. hb_bytes_t has hash() defined so that should work without any additional changes. Then for each string you can check if it already exists, and grab it's index for re-use easily. Should simplify this change.